src/NHibernate/Engine/Query/ParameterParser.cs

```diff
@@ -48,47 +48,47 @@ public static void Parse(string sqlString, IRecognizer recognizer)
 
 			int stringLength = sqlString.Length;
 			bool inQuote = false;
-			bool afterNewLine = false;
 			for (int indx = 0; indx < stringLength; indx++)
 			{
 				int currentNewLineLength;
 
-				// check comments
-				if (indx + 1 < stringLength && sqlString.Substring(indx,2) == "/*")
+				// check comments, unless in quote
+				if (!inQuote)
 				{
-					var closeCommentIdx = sqlString.IndexOf("*/", indx + 2, StringComparison.Ordinal);
-					recognizer.Other(sqlString.Substring(indx, (closeCommentIdx- indx)+2));
-					indx = closeCommentIdx + 1;
-					continue;
-				}
-				
-				if (afterNewLine && (indx + 1 < stringLength) && sqlString.Substring(indx, 2) == "--")
```

Comment:
So, this parser only accounted for single line comments starting at the beginning of the line. Any thought on why was that? Just a false assumption of who has coded that? Or some db actually handling it that way?

Reply:
I think it was a false assumption.